### ISSUE TITLE
G attribute transform errors in Line Chart when resizing browser's window

### ISSUE THREAD

**Issue by mariela-plaza** [plouc/nivo#2111]
First of all, thank you for this amazing library. I've been using it at work and the results have been pretty great.

Lately, I've been working with the Line Chart and I noticed that if I implement it to be responsive and change it's width when the window changes sizes, I get a bunch of errors in the console regarding the g attribute transform:
![g-attrbitute-errors](https://user-images.githubusercontent.com/24737923/185341904-cbdb5009-c444-4d8b-aa96-3453f31262a5.png)

To make sure it wasn't just me, I opened the Dev Tools in the nivo page of the Line Chart and noticed it also happens there:

https://user-images.githubusercontent.com/24737923/185344267-9a4290e2-99f3-41c2-bc79-94fd7ed2601e.mp4

I tested this in multiple browsers and they all deliver the same type of error. The graph itself works correctly, but the amount of errors in the console is pretty big, so I thought it would be a good idea to report it.

Thank you for all of the great work you've done with the library!

**Comment by mariela-plaza** [comment]
bump!

**Comment by moutafatin1** [comment]
bump!

**Comment by Brevan4** [comment]
bump

**Comment by KarolinaCharlinska** [comment]
bump

**Comment by anarguez** [comment]
bump

**Comment by KarolinaCharlinska** [comment]
Try to turn off the animation. It works for me.

`animate={false}`

**Comment by BigPapoo** [comment]
Switch off animation works... but keeping anims on would be nice :-)

**Comment by svn1991** [comment]
is there a way to make it work with the animate={true} ? The animation is actually nice and makes the interaction look smooth.

**Comment by FrancisRoc** [comment]
This is still happening in version 0.87.0 released recently, please reopen!

**Comment by plouc** [comment]
@FrancisRoc, it still sometimes happens on the demo website as well, so I guess it's still valid.

**Comment by FrancisRoc** [comment]
@plouc Very responsive to my comment that great, thanks! I initially thought that in my case it's cause I dynamically change the `tickRotation` based on screen resize. But I used memo to make sure I don't rerender if the props didn't change and I was still able to reproduce. 
It seems in some cases when we resize we end up having one of the ticks on the bottom axis to be 'outside' the chart by a very small value which switches to exponential notation and it seems that it creates the errors ? Doesn't happen all the time and I couldn't find a specific pattern. Hope that is helpful in any way 🙏🏼 
![image](https://github.com/plouc/nivo/assets/26251931/9b7d6583-ce31-4700-853a-16edf90ce821)

**Comment by Coreeze** [comment]
still happens. version `0.87.0`

**Comment by plouc** [comment]
@Coreeze, I don't think this kind of reasoning works, especially for open source projects, rather than complaining that no one fixes it, maybe you could try to contribute.

I won't remove animations from the lib because of a bug which only appears in certain situations, plus, while it shows in the console, it's not crashing, it's not preventing users from seeing the chart. And animations clearly work for many other chart types.

Seeing such comments is really demotivating, if you're not happy with the lib or just want a solution with full support, feel free to use something else, but please refrain from making such comments, this is not constructive at all.

**Comment by plouc** [comment]
> @plouc Very responsive to my comment that great, thanks! I initially thought that in my case it's cause I dynamically change the `tickRotation` based on screen resize. But I used memo to make sure I don't rerender if the props didn't change and I was still able to reproduce. It seems in some cases when we resize we end up having one of the ticks on the bottom axis to be 'outside' the chart by a very small value which switches to exponential notation and it seems that it creates the errors ? Doesn't happen all the time and I couldn't find a specific pattern. Hope that is helpful in any way 🙏🏼 For the screenshot: I used `renderTick` to see the properties so the screenshot is just one of the `renderTick` printed props ![image](https://private-user-images.githubusercontent.com/26251931/330126638-9b7d6583-ce31-4700-853a-16edf90ce821.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYyNDcxNDksIm5iZiI6MTcxNjI0Njg0OSwicGF0aCI6Ii8yNjI1MTkzMS8zMzAxMjY2MzgtOWI3ZDY1ODMtY2UzMS00NzAwLTg1M2EtMTZlZGY5MGNlODIxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIwVDIzMTQwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI5ZTBiYjZiMjM0NDg5ZTAyNWQwOTY1ZDlmMDNjMDk3YmI4Zjc4MjE3MGI0MjE0ODU0MTgyNDZiN2FlMTUyMjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.oefSD0qAnX5a2LTjaNXrwBVvmQKArs5oWzdVMuq4KZo)

That's helpful, it seems like this is not due to animation, because the static `x` value has the issue already, we probably need  to limit the precision of the computed position.

**Comment by Coreeze** [comment]
> @Coreeze, I don't think this kind of reasoning works, especially for open source projects, rather than complaining that no one fixes it, maybe you could try to contribute. 
> 
> I won't remove animations from the lib because of a bug which only appears in certain situations, plus, while it shows in the console, it's not crashing, it's not preventing users from seeing the chart. And animations clearly work for many other chart types.
> 
> Seeing such comments is really demotivating, if you're not happy with the lib or just want a solution with full support, feel free to use something else, but please refrain from making such comments, this is not constructive at all.

@plouc hey, you are right. i was just frustrated. i apologize and thank you for building and supporting this library :)

**Comment by stefanrows** [comment]
This seems to be still the case? It's even present on the actual Nivo site. Is there any way we can get this resolved? I'd love to use the chart, but not with these log spams.

EDIT: Fully willing to contribute btw, just hit me up!

<img width="2556" height="1203" alt="Image" src="https://github.com/user-attachments/assets/db6998d7-71f4-4556-8abc-a74eebebb761" />
